File: src/net/eduard/api/server/GeneratorSystem.java
package net.eduard.api.server;

import org.bukkit.Location;
import org.bukkit.block.Block;
import org.bukkit.entity.EntityType;
import org.bukkit.inventory.ItemStack;

public interface GeneratorSystem {
	
	public ItemStack getGenerator(EntityType type);
	
	public EntityType getGeneratorType(ItemStack item);

	public int getGeneratorStack(Location location);

	public EntityType getGeneratorType(Location location);
	
	public boolean isGenerator(ItemStack item);
}

File: src/main/java/net/eduard/api/server/GeneratorSystem.kt
package net.eduard.api.server

import org.bukkit.Location
import org.bukkit.block.Block
import org.bukkit.entity.EntityType
import org.bukkit.inventory.ItemStack

interface GeneratorSystem {

    interface Generator{

        fun getStack() : Int
        fun getType() : EntityType
        fun modifyType(type: EntityType)
        fun newStack(stack : Int)

    }
    fun getGeneratorAtLocation(location: Location): Generator

    fun getGenerator(type: EntityType): ItemStack

    fun getGeneratorType(item: ItemStack): EntityType

    fun getGeneratorStack(location: Location): Int

    fun getGeneratorType(location: Location): EntityType


    fun isGenerator(item: ItemStack): Boolean
}
